File: data_agent/tests/integration/test_health.py
from fastapi.testclient import TestClient

from app.main import app

client = TestClient(app)


def test_health_endpoint() -> None:
    response = client.get("/health")

    assert response.status_code == 200
    assert response.json() == {
        "status": "ok",
        "service": "maranello-ai-data-agent",
        "environment": "development",
    }


def test_root_endpoint() -> None:
    response = client.get("/")

    assert response.status_code == 200
    assert response.json() == {
        "service": "maranello-ai-data-agent",
        "status": "running",
        "environment": "development",
    }

File: backend/tests/integration/test_health.ts
import request from "supertest";
import { describe, expect, it } from "vitest";

import { app } from "../../src/app.js";


describe("Backend health endpoints", () => {
  it("returns the backend health status", async () => {
    const response = await request(app)
      .get("/health")
      .expect(200);

    expect(response.body).toEqual({
      status: "ok",
      service: "maranello-ai-backend",
      environment: "test",
    });
  });

  it("returns the backend root status", async () => {
    const response = await request(app)
      .get("/")
      .expect(200);

    expect(response.body).toEqual({
      service: "maranello-ai-backend",
      status: "running",
      environment: "test",
    });
  });
});